lib/health.js

```diff
@@ -48,17 +48,35 @@ class HealthCheck {
     };
   }  
 
+  http(request, response) {
```

Comment:
This doesn't follow the pattern established by the other handlers.   `http` should return the function currently known as `http`, which should be anonymous.  The additional layer for async is still needed, so it's 
```javascript
http() {
  return (request, response) => {
    // L52-65
  };
}
```

And then the usage needs to be 

```javascript
const check = healthcheck();
const handler = check.http();
http.createServer( (request, response) => {
  const isHealthCheckCall = handler(request, response);
  if (isHealthCheckCall) return;
   response.end("HELLO! This is pretty amaziiiiing");
}).listen(3535);
```

Reply:
So, I agree that consistency would be nice, and I like where you are heading but the specific solution you propose won't work since implementation of other functions return an async function and your returns regular one. Koa and Express can take async function as their middleware, Node's vanilla http has no concept of middleware and if we try doing `const isHealthCheckCall = await handler(request, response);` than its surrounding context will have to become async and that can make usage of Maikai really awkward.

Which is why I decided to sacrifice consistency for the sake of "it should be idiomatic for the use-case in which it is used". For somebody using this with vanilla node http - do they care about
1. whether somebody else in Koa uses it the same way
or 
2. whether their usage is close to what they would be doing without healthcheck middleware?

and the decision in the current design was made in favor of #2. 

Thoughts?

Reply:
Here's a possible compromise, if http() returns a non-async function with the following implementation:

```javascript
  http() {
    return (request, response) => {
      const requestPath = request.url;
      if (requestPath !== this.getHealthUri()) return false;

      const self = this;

      (async function () {
        const responseContent = await self.healthResponse(self.opts);
        const httpCode = (response.status === STATUS_ERR ) ? 503 : 200;
        
        response.writeHead(httpCode, { 'Content-Type': 'application/health+json; charset=utf-8' });
        response.end(JSON.stringify(responseContent));
      })();

      return true;
    };
  }
```

then the calling code does become similar to wrote you proposed:

```javascript
const http = require('http');
const healthcheck = require('../');

http.createServer( (request, response) => {
  const check = healthcheck();
  const checkHandler = check.http();
  const isHealthCheckCall = checkHandler(request, response);
  if (isHealthCheckCall) return;

  response.end("HELLO! This is pretty amaziiiiing");
}).listen(3535);
```

Reply:
Is that what you meant, to begin with?

Reply:
It was exactly what I meant to begin with, yes!    Perhaps I could have been slightly more explicit.  :)